torch/distributions/bernoulli.py

```diff
@@ -72,7 +72,8 @@ def param_shape(self):
 
     def sample(self, sample_shape=torch.Size()):
         shape = self._extended_shape(sample_shape)
-        return torch.bernoulli(self.probs.expand(shape))
+        x = torch.bernoulli(self.probs.expand(shape))
```

Comment:
Use:

```
with torch.no_grad():
  return torch.bernoulli(self.probs.expand(shape))
```

instead of detaching at the end

Reply:
This will also make it faster, because `hasattr` is quite expensive

Reply:
Thanks, I'll start using `torch.no_grad()`.